fence/agents/azure_arm/fence_azure_arm_network.py

```diff
@@ -0,0 +1,215 @@
+#!@PYTHON@ -tt
+
+import sys, re, pexpect
+import logging
+import atexit
+sys.path.append("@FENCEAGENTSLIBDIR@")
+from fencing import *
+from fencing import fail, fail_usage, EC_TIMED_OUT, run_delay
+import time
+import azure_fence_lib
+
+#import ptvsd
+
+def get_azure_config(options):
+    config = azure_fence_lib.AzureConfiguration()
+
```

Comment:
You should be able to get a bunch of this information from IMDS dynamically.  That way you don't need to have the user supply all this via configuration.  See https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service, and in particular, https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service#instance-metadata-data-categories.

Reply:
Great idea! I made resourceGroup and subscriptionId optional.